x/bank/keeper/keeper_test.go

```diff
@@ -0,0 +1,279 @@
+package keeper_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdksimapp "github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	bankTypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/certikfoundation/shentu/v2/simapp"
+	vesting "github.com/certikfoundation/shentu/v2/x/auth/types"
+	"github.com/certikfoundation/shentu/v2/x/bank/keeper"
+	"github.com/certikfoundation/shentu/v2/x/bank/types"
+)
+
+var (
+	acc1 = sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+	acc2 = sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+	acc3 = sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+	acc4 = sdk.AccAddress(ed25519.GenPrivKey().PubKey().Address().Bytes())
+)
+
+// shared setup
+type KeeperTestSuite struct {
+	suite.Suite
+
+	address     []sdk.AccAddress
+	app         *simapp.SimApp
+	ctx         sdk.Context
+	queryClient types.MsgServer
+	params      types.AccountKeeper
+	keeper      keeper.Keeper
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	suite.app = simapp.Setup(false)
+	suite.ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{})
+	suite.keeper = suite.app.BankKeeper
+	suite.params = suite.app.AccountKeeper
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.app.InterfaceRegistry())
+	types.RegisterMsgServer(queryHelper, suite.queryClient)
+	suite.queryClient = &types.UnimplementedMsgServer{}
+
+	for _, acc := range []sdk.AccAddress{acc1, acc2, acc3, acc4} {
+		err := sdksimapp.FundAccount(
+			suite.app.BankKeeper,
+			suite.ctx,
+			acc,
+			sdk.NewCoins(
+				sdk.NewCoin("ctk", sdk.NewInt(1000)), // 1,000 CTK
+			),
+		)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	suite.address = []sdk.AccAddress{acc1, acc2, acc3, acc4}
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+func (suite *KeeperTestSuite) TestSendCoins() {
+	type args struct {
+		fromAddr   sdk.AccAddress
+		toAddr     sdk.AccAddress
+		amount     int64
+		accBalance int64
+	}
+
+	type errArgs struct {
+		shouldPass bool
+		contains   string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}{
+		{"Operator(1) Create: first send test case if coins is not greater than total amount",
```

Comment:
also let's make the test names a complete sentence or at least have correct grammar